### PR TITLE
feat/gh actions pipelines

### PR DESCRIPTION
**Pull request by mmpetarpeshev**
Before we merge , we need secrets to be add to the repo and dockerhub repo to be done.
But is done for review , I will handle the other things in meantime.

### CODE REVIEW

**Comment by marc0olo** [comment]
added also @dincho to have a quick look if he finds some time :)

@mmpetarpeshev do you need sth. from me?

**Comment by mmpetarpeshev** [comment]
> added also @dincho to have a quick look if he finds some time :)
> 
> @mmpetarpeshev do you need sth. from me?

Yes ,if you have perms. I will need to allow the repository to have access to github secrets.

**Comment by marc0olo** [comment]
> Yes ,if you have perms. I will need to allow the repository to have access to github secrets.

done. do we also need that for the backend repo?

File: nginx/default.conf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+server {
+  location / {
+      root /usr/share/nginx/html;
+      try_files $uri $uri/ /index.html;
+  }
+}
+
+server {
+  server_name aepp.dex.superhero.com;
```
Comment (thepiwo) [inline nginx/default.conf]: copy paste leftover?

File: nginx/nginx.conf [hunk 2]
```diff
@@ -0,0 +1,19 @@
+user  nginx;
```
Comment (thepiwo) [inline nginx/nginx.conf]: do we really need this config file?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+FROM node:18.7.0 as builder
+
+RUN mkdir -p /home/node/app
+WORKDIR /home/node/app
+COPY package*.json /home/node/app/
+RUN chown -R node:node /home/node/
+
+USER node
+
+RUN yarn install
```
Comment (thepiwo) [inline Dockerfile]: lets use frozen lockfile here

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,17 @@
+FROM node:18.7.0 as builder
+
+RUN mkdir -p /home/node/app
+WORKDIR /home/node/app
+COPY package*.json /home/node/app/
```
Comment (thepiwo) [inline Dockerfile]: this is not needed with * but we need yarn.lock additionally

**Comment by mmpetarpeshev** [comment]
> > Yes ,if you have perms. I will need to allow the repository to have access to github secrets.
> 
> done. do we also need that for the backend repo?

Yes and thank you.

**Comment by marc0olo** [comment]
> Yes and thank you.

done

File: nginx/default.conf (hunk 1 above)
Comment (mmpetarpeshev) [reply nginx/default.conf]: Correct , I started to remove it and something else popup.

File: Dockerfile (hunk 3 above)
Comment (mmpetarpeshev) [reply Dockerfile]: Okey, thats why I asked you , can you please let me know , what exactly is needed for the install and build.

File: Dockerfile (hunk 4 above)
Comment (mmpetarpeshev) [reply Dockerfile]: Great , will change them.

File: Dockerfile (hunk 3 above)
Comment (thepiwo) [reply Dockerfile]: yes, this should be adjusted

File: Dockerfile (hunk 4 above)
Comment (thepiwo) [reply Dockerfile]: thanks

**Review by thepiwo — APPROVED** [review]
lgtm

**Pull request merged**